app/main.py

```diff
@@ -1,3 +1,79 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, distance: Distance) -> None:
+        self.km = distance
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance or int) -> Distance:
+        if isinstance(other, Distance):
+            self.km = self.km + other.km
+        else:
+            self.km = self.km + other.real
+        return Distance(distance=self.km)
+
+    def __iadd__(self, other: Distance or int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            # return Distance(distance=self.km + other.real)
+            self.km += other.real
+        # return Distance(distance=self.km)
+        return self
+
+    def __mul__(self, other: Distance or int) -> Distance or None:
+        if isinstance(other, Distance):
+            return None
+        else:
+            return Distance(distance=self.km * other.real)
```

Comment:
`other` can not be of type `Distance` in this method, because if you multiply km * km you get square km. You should always return a Distnace instance in this method